Lib/imaplib.py

```diff
@@ -588,6 +648,48 @@ def getquotaroot(self, mailbox):
         return typ, [quotaroot, quota]
 
 
+    def idle(self, duration=None):
+        """Return an Idler: an iterable context manager for the IDLE command
+
+        :param duration:  Maximum duration (in seconds) to keep idling,
+                          or None for no time limit.
+                          To avoid inactivity timeouts on servers that impose
+                          them, callers are advised to keep this <= 29 minutes.
+                          See the note below regarding IMAP4_stream on Windows.
+        :type duration:   int|float|None
```

Comment:
Since we don't use autodoc on the sources, I don't think we should be that verbose on the Sphinx syntax. Considering the other commands, I would use the same docstring style.